packages/pluggableWidgets/app-events-native/src/AppEvents.tsx

```diff
@@ -43,6 +43,10 @@ export class AppEvents extends Component<Props> {
     }
 
     componentWillUnmount(): void {
+        if (this.props.onUnloadAction && this.props.onUnloadAction.canExecute) {
```

Comment:
Can I request you to kindly move this implementation for Funcional component?

Reply:
Do you mean in an effect hook? Even though the onload is using the lifecycle method?

Reply:
Its fine, we will move the whole widget then to FC after this MR